onnxruntime/core/providers/webnn/builders/impl/reduction_op_builder.cc

```diff
@@ -134,8 +134,9 @@ bool ReductionOpBuilder::IsOpSupportedImpl(const InitializedTensorSet& initializ
     return false;
 
   const auto& op_type = node.OpType();
+  const auto& axes_name = input_defs[1]->Name();
   // If the optional input 'axes' is provided, it must be an initializer.
-  if (input_defs.size() > 1 && !Contains(initializers, input_defs[1]->Name())) {
+  if (input_defs.size() > 1 && !axes_name.empty() && !Contains(initializers, axes_name)) {
```

Comment:
If it's possible for `input_defs.size to be <=1`, then don't we need a check on line 137 before accessing [1]?

If we use this often enough, introducing a helper would reduce the number of manual size checks and brittleness. e.g.:

```
  std::string_view axes_name = GetTensorName(input_defs, 1);
  // If the optional input 'axes' is provided, it must be an initializer.
  if (!axes_name.empty() && !Contains(initializers, axes_name)) {
```

`GetTensorName` does the bounds check, returning `""` if not present so that optional tensors are treated consistently in either of the two representations. Then line 139 doesn't need to check the size anymore.

Reply:
Thanks! Added `GetTensorName`, code is simplified now.